tools/README.md

```diff
@@ -28,27 +37,47 @@ When used with ``--map`` and a single elf file, generate a map consisting of mem
 With ``--qdirstat`` and a single elf file, generate a [qdirstat](https://github.com/shundhammer/qdirstat) compatible cache file which can be loaded to inspect memory utilization interactively in a treemap.
 
 This assumes the running firmware generating the dump and the elf file are the same.
-Requires Python3 and the [pyelftools](https://github.com/eliben/pyelftools) module.
+Requires the [pyelftools](https://github.com/eliben/pyelftools) module.
+
+
+## Crash dump handling
+
+### ``dump_crash``
+
+Dump the content of the last crash dump on MK3+ printers using D21.
+Requires ``printcore`` from [Pronterface].
 
 ### ``dump2bin``
 
 Parse and decode a memory dump obtained from the D2/D21/D23 g-code into readable metadata and binary. The output binary is padded and extended to fit the original address range.
 
 ### ``xfimg2dump``
 
-Extract a crash dump from an external flash image and output the same format produced by the D21 g-code.
-
-### ``update_eeprom``
+Extract a crash dump from an external flash image and output the same format produced by the D21 g-code. Requires python 3.
 
-Given one EEPROM dump, convert the dump to update instructions that can be sent to a printer.
 
-Given two EEPROM dumps, produces only the required instructions needed to update the contents from the first to the second. This is currently quite crude and assumes dumps are aligned (starting from the same address or same stride).
-
-Optionally writes the instructions to the specified port (requires ``printcore`` from [Pronterface]).
+## Serial handling
 
 ### ``noreset``
 
 Set the required TTY flags on the specified port to avoid reset-on-connect for *subsequent* requests (issuing this command might still cause the printer to reset).
 
 
+## Temperature analysis
+
+### ``tml_decode``
+
+Decode (or plot) the temperature model trace from a serial log file.
+
+The TML trace needs to be enabled by issuing "M155 S1 C3" and "D70 S1" to the printer, generally followed by a temperature model calibration request "M310 A F0".
+
+The parser is not strict, and will consume most serial logs with/without timestamps.
+
+By default the decoded trace is written to the standard output as a tab-separated table. If `--plot` is used, produce a graph into the requested output file instead:
+
+    ./tml_decode -o graph.png serial.log
```

Comment:
Typo the plot argument is `-p`

Reply:
Indeed. I also noticed another one: https://github.com/prusa3d/Prusa-Firmware/pull/3943